IdentityCore/src/broker_operation/request/silent_token_request/MSIDBrokerOperationSilentTokenRequest.m

```diff
@@ -97,5 +103,49 @@ - (NSDictionary *)jsonDictionary
     return json;
 }
 
+- (NSString *)fullRequestThumbprint
+{
+    return [MSIDThumbprintCalculator calculateThumbprint:[self jsonDictionary]
+                                            filteringSet:[MSIDBrokerOperationSilentTokenRequest fullRequestThumbprintExcludeParams]
+                                       shouldIncludeKeys:NO];
+}
+
+- (NSString *)strictRequestThumbprint
+{
+    return [MSIDThumbprintCalculator calculateThumbprint:[self jsonDictionary]
+                                            filteringSet:[MSIDBrokerOperationSilentTokenRequest strictRequestThumbprintIncludeParams]
+                                       shouldIncludeKeys:YES];
+}
+
+
++ (NSSet *)fullRequestThumbprintExcludeParams
+{
+    static dispatch_once_t once_token;
+    static NSSet *excludeSet;
+    
+    dispatch_once(&once_token, ^{
+        excludeSet = [NSSet setWithArray:@[MSID_ACCOUNT_DISPLAYABLE_ID_JSON_KEY, //since we already use home account id, this is redundant
+                                           MSID_BROKER_CLIENT_CAPABILITIES_KEY,
+                                           MSID_BROKER_CLIENT_VERSION_KEY,
+                                           MSID_BROKER_CLIENT_APP_VERSION_KEY,
+                                           MSID_BROKER_CLIENT_APP_NAME_KEY,
+                                           MSID_BROKER_CORRELATION_ID_KEY]];
+    });
+    return excludeSet;
+    
+}
+
++ (NSSet *)strictRequestThumbprintIncludeParams
+{
+    static dispatch_once_t once_token;
+    static NSSet *includeSet;
+    
+    dispatch_once(&once_token, ^{
+        includeSet = [NSSet setWithArray:@[MSID_CLIENT_ID_JSON_KEY, MSID_SCOPE_JSON_KEY, MSID_AUTHORITY_URL_JSON_KEY, MSID_ACCOUNT_HOME_ID_JSON_KEY]];
```

Comment:
Should we account for claims and instanceAware?

Reply:
They are included in the full request parameter calculation, but for strict request thumbprint, 
client_id, scope, authority url, and home account should suffice. This is also in line with the original spec: https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?version=GBdev&path=%2FService%20protection%2FIntial%20set%20of%20protection%20measures.md&_a=preview&anchor=common-test-cases